core/src/main/scala/io/finch/internal/ToResponse.scala

```diff
@@ -46,11 +46,22 @@ object ToResponse extends LowPriorityToResponseInstances {
 
   implicit def asyncBufToResponse[CT <: String](implicit
     w: Witness.Aux[CT]
-  ): Aux[AsyncStream[Buf], CT] = instance { a =>
+  ): Aux[AsyncStream[Buf], CT] = asyncStreamResponseBuilder(identity)
+
+  implicit def asyncStreamToResponse[A, CT <: String](implicit
+    e: Encode.Aux[A, CT],
+    w: Witness.Aux[CT]
+  ): Aux[AsyncStream[A], CT] = asyncStreamResponseBuilder(data => e(data) concat Buf.Utf8("\n"))
+
+  private def asyncStreamResponseBuilder[A, CT <: String](writer: A => Buf)(implicit
```

Comment:
Won't work because there is a difference between buffer streaming and JSON streaming.

- In case of JSON streaming we add EOL symbol
- In case of buffer streaming we cant add EOL since it could completly break data in stream

Different behaviours